js/src/views/Contract/Queries/queries.js

```diff
@@ -34,27 +47,79 @@ export default class Queries extends Component {
 
     const queries = contract.functions
       .filter((fn) => fn.constant)
-      .sort(this._sortEntries)
-      .map((fn) => {
-        return (
-          <div
-            key={ fn.signature }
-            className={ styles.method }>
-            { fn.name }
-          </div>
-        );
-      });
+      .sort(this._sortEntries);
+
+    const noInputQueries = queries
+      .slice()
+      .filter((fn) => fn.inputs.length === 0)
+      .map((fn) => this.renderQuery(fn));
+
+    const withInputQueries = queries
+      .slice()
+      .filter((fn) => fn.inputs.length > 0)
+      .map((fn) => this.renderInputQuery(fn));
 
     return (
       <Container>
         <ContainerTitle title='queries' />
         <div className={ styles.methods }>
-          { queries }
+          { noInputQueries }
+          { withInputQueries }
         </div>
       </Container>
     );
   }
 
+  renderInputQuery (fn) {
+    const { inputs, outputs } = fn;
+    const { contract } = this.props;
+
+    return (
+      <div key={ fn.signature }>
+        <InputQueries
+          inputs={ inputs }
+          outputs={ outputs }
+          name={ fn.name }
+          contract={ contract }
+        />
+      </div>
+    );
+  }
+
+  renderQuery (fn) {
+    const { values } = this.props;
+
+    return (
+      <div
+        key={ fn.signature }>
+        <Card style={ {
```

Comment:
Visually needs some work. Probably go with transparency of the background as well. At the same time when displayed in a column, the cards needs to be the same height. Looks a bit jaggerd :( (I know, not easy to do)

In addition, CSS in .css, even if it comes with an !important.